src/components/ReportingConfig/ReportingConfigForm.jsx

```diff
@@ -0,0 +1,436 @@
+
+//  All the fields in this form that need to be validated can be added here.
+const REQUIRED_FIELDS = [
+  'hourOfDay',
+];
+// for the email delivery method
+const REQUIRED_EMAIL_FIELDS = [
+  'email',
+];
+// for the sftp delivery mothod
+const REQUIRED_SFTP_FIELDS = [
+  'sftpPort',
+  'sftpHostname',
+  'sftpUsername',
+  'sftpPassword',
+  'sftpFilePath',
+];
+/**
+ * Validates this form. If the form is valid, it will return true.
+ * Otherwise, it returns the invalid fields.
+ * @param {FormData} formData
+ * @param {[String]} requiredFields
+ */
+const validateReportingForm = (formData, requiredFields) => {
+  const invalidFields = requiredFields
+    .filter(field => !formData.get(field))
+    .reduce((prevFields, currField) => ({ ...prevFields, [currField]: true }), {});
+  return isEmpty(invalidFields) || invalidFields;
+};
+
+/**
+ * Transform the form data from camelCase back into snake case to
+ * prepare it to be sent down the wire.
+ * @param {*} formData
+ */
+const transformFormData = (formData) => {
+  const transformedData = new FormData();
+  [...formData.entries()]
+    .forEach(entry => (
+      transformedData.append(snakeCase(entry[0]), entry[1])
+    ));
+  return transformedData;
+};
+
+/**
+ * Used as both the form to create and edit an existing reporting configuration.
+ * @param {Object} config -> incoming configuration, if it exists
+ */
+const ReportingConfigForm = ({
+  config,
+  createConfig,
+  updateConfig,
+}) => {
+  const [frequency, setFrequency] = useState(config ? config.frequency : 'monthly');
+  const [deliveryMethod, setDeliveryMethod] = useState(config ? config.deliveryMethod : 'email');
+  const [invalidFields, setInvalidField] = useState({});
+  const [active, setActive] = useState(config ? config.active : false);
+  const [submitState, setSubmitState] = useState('default');
+  return (
+    <form
+      onSubmit={async (e) => {
+        e.preventDefault();
+        setSubmitState('pending');
+        const formData = new FormData(e.target);
+        const requiredFields = REQUIRED_FIELDS;
+        if (deliveryMethod === 'email') {
+          requiredFields.push(...REQUIRED_EMAIL_FIELDS);
+        }
+        if (deliveryMethod === 'sftp') {
+          requiredFields.push(...REQUIRED_SFTP_FIELDS);
+        }
+        const valid = validateReportingForm(formData, requiredFields);
+        if (valid === true) {
+          // we need to transform the formdata back to snake_case before sending it
+          const transformedData = transformFormData(formData);
+          // if this is not a new config, add the uuids
+          if (config) {
+            transformedData.append('uuid', config.uuid);
+            transformedData.append('enterprise_customer_id', config.enterpriseCustomer.uuid);
+            const err = await updateConfig(transformedData, config.uuid);
+            if (err) {
+              setSubmitState('error');
```

Comment:
nit: Consider making states like this that end up being used more than once a Constant.  It makes the code maintenance easier.

i.e. 'default', 'error', 'complete', etc.

Reply:
handled in the refactor